### PR TITLE
LayerUpdater fix

### PR DESCRIPTION
**Pull request by pomadchin**
* leftOuterJoin problem
* layer key bounds were not updating in metadata table

### CODE REVIEW

File: accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerUpdater.scala [hunk 1]
```diff
@@ -65,15 +63,13 @@ class AccumuloLayerUpdater(
...
-            case None =>
-              layerTile
-          }
-      }
+        .fullOuterJoin(rdd)
+        .mapValues {
+          case (Some(layerTile), Some(updateTile)) => mergeFunc(layerTile, updateTile)
+          case (Some(layerTile), _) => layerTile
+          case (_, Some(updateTile)) => updateTile
+          case _ => throw new LayerUpdateError(id, "No value(s) in your existing and new layer RDDs")
```
Comment (lossyrob) [inline accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerUpdater.scala]: The exception is weird here and not every going to get hit...maybe better to

```scala
val updatedRdd: RDD[(K, V)] =
  existingTiles
    .fullOuterJoin(rdd)
    .flatMapValues {
      case (Some(layerTile), Some(updateTile)) => Some(mergeFunc(layerTile, updateTile))
      case (Some(layerTile), _) => Some(layerTile)
      case (_, Some(updateTile)) => Some(updateTile)
      case _ => None
    }
```

Comment (pomadchin) [inline accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerUpdater.scala]: I like that, flatMap flew out of my mind ._.

**Comment by lossyrob** [comment]
+1 after travis

**Comment by lossyrob** [comment]
@pomadchin might think about squashing a lot of these codestyle commits so that there are clearer commit messages. Also when I backport I have to copy all these commits back, not a lot of changes for cherry-picking 10 commits :)

Don't sweat it on this one, after travis passes I want to merge this and backport, just something to note moving forward

**Comment by pomadchin** [comment]
@lossyrob i thought about it, sorry for that ._. ; can revert them and / or to make a clean pr, what do you think about it?

**Comment by lossyrob** [comment]
no worries about this one, just want travis to pass and get this into 0.10.1

**Pull request merged**